### PR TITLE
OWIN sameSite

### PR DESCRIPTION
**Pull request by Rick-Anderson**
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/samesite/owin-samesite?branch=pr-en-us-279)
Fixes https://github.com/aspnet/AspNetCore.Docs/issues/16027

### CODE REVIEW

File: aspnet/toc.yml [hunk 1]
```diff
@@ -3763,6 +3763,10 @@
       href: aspnet/overview/owin-and-katana/owin-middleware-in-the-iis-integrated-pipeline.md
     - name: Host OWIN in an Azure Worker Role
       href: aspnet/overview/owin-and-katana/host-owin-in-an-azure-worker-role.md
+- name: SameSite
+  items:
+    - name: OWIN SameSite
+    uid: SameSite/owin-samesite
```
Comment (scottaddie) [inline aspnet/toc.yml]:
```suggestion
      uid: SameSite/owin-samesite
```

File: aspnet/toc.yml [hunk 2]
```diff
@@ -3763,6 +3763,10 @@
       href: aspnet/overview/owin-and-katana/owin-middleware-in-the-iis-integrated-pipeline.md
     - name: Host OWIN in an Azure Worker Role
       href: aspnet/overview/owin-and-katana/host-owin-in-an-azure-worker-role.md
+- name: SameSite
+  items: 
+    - name: OWIN SameSite
+      uid: sametite/owin-samesite
```
Comment (scottaddie) [inline aspnet/toc.yml]:
```suggestion
      uid: owin-samesite
```

File: aspnet/SameSite/owin-samesite.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+# SameSite cookies and the Open Web Interface for .NET (OWIN)
+
+By [Rick Anderson](https://twitter.com/RickAndMSFT)
+
+SameSite is an [IETF](https://ietf.org/about/) draft designed to provide some protection against cross-site request forgery (CSRF) attacks. The [SameSite 2019 draft](https://tools.ietf.org/html/draft-west-cookie-incrementalism-00):
+
+* Treats cookies as `SameSite=Lax` by default.
+* States cookies that explicitly assert `SameSite=None` in order to enable cross-site delivery should be marked as `Secure`.
+
+`Lax` works for most app cookies. Some forms of authentication like [OpenID Connect](https://openid.net/connect/) (OIDC) and [WS-Federation](https://auth0.com/docs/protocols/ws-fed) default to POST based redirects. The POST based redirects trigger the SameSite browser protections, so SameSite is disabled for these components. Most [OAuth](https://oauth.net/) logins are not affected due to differences in how the request flows.
```
Comment (Tratcher) [inline aspnet/SameSite/owin-samesite.md]: All other components do not set SameSite by default and use the clients default behavior (old or new).

File: aspnet/SameSite/owin-samesite.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+The `None` parameter causes compatibility problems with clients that implemented the prior [2016 draft standard](https://tools.ietf.org/html/draft-west-first-party-cookies-07) (for example, iOS 12). See [Supporting older browsers](#sob) in this document.
+
+Each OWIN component that emits cookies needs to decide if SameSite is appropriate.
+
+## API usage with SameSite
+
+[Microsoft.Owin](https://www.nuget.org/packages/Microsoft.Owin/) uses a nullable SameSite field in most places.
```
[the remark this answers is not in the document]
Comment (Tratcher) [reply aspnet/SameSite/owin-samesite.md]: How do we get the API docs updated for 4.1? These docs don't even say what version they're from.
https://docs.microsoft.com/en-us/previous-versions/aspnet/dn253692(v=vs.113)

[the remark this answers is not in the document]
Comment (Rick-Anderson) [reply aspnet/SameSite/owin-samesite.md]: > How do we get the API docs updated for 4.1? These docs don't even say what version they're from.
https://docs.microsoft.com/en-us/previous-versions/aspnet/dn253692(v=vs.113)

@BillWagner ?

File: aspnet/SameSite/owin-samesite.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+```csharp
+owinContext.Response.Cookies.Append("My Key", "My Value", new CookieOptions()
+{
+    SameSite = SameSiteMode.Lax
+});
+```
+
+The following APIs use SameSite:
+
+* <xref:System.Web.SameSiteMode>
```
Comment (Tratcher) [inline aspnet/SameSite/owin-samesite.md]:
```suggestion
* <xref:Microsoft.Owin.SameSiteMode>
```

File: aspnet/SameSite/owin-samesite.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+* [CookieAuthenticationOptions Class](/previous-versions/aspnet/dn385599(v%3Dvs.113)) <!-- CookieAuthenticationOptions.CookieSameSite not published -->
+* [CookieAuthenticationOptions.CookieSameSite](https://github.com/aspnet/AspNetKatana/blob/dev/src/Microsoft.Owin.Security.Cookies/CookieAuthenticationOptions.cs#L68-#L72)
+* [ICookieManager](/previous-versions/aspnet/dn800238(v%3Dvs.113))
+* [SystemWebCookieManager](https://github.com/aspnet/AspNetKatana/blob/dev/src/Microsoft.Owin.Host.SystemWeb/SystemWebCookieManager.cs)
+* [SystemWebChunkingCookieManager](https://github.com/aspnet/AspNetKatana/blob/dev/src/Microsoft.Owin.Host.SystemWeb/SystemWebChunkingCookieManager.cs)
+* [CookieAuthenticationOptions.CookieManager](https://github.com/aspnet/AspNetKatana/blob/dev/src/Microsoft.Owin.Security.Cookies/CookieAuthenticationOptions.cs#L143-#AL148)
+* [OpenIdConnectAuthenticationOptions.CookieManager](https://github.com/aspnet/AspNetKatana/blob/dev/src/Microsoft.Owin.Security.OpenIdConnect/OpenIdConnectAuthenticationOptions.cs#L315-#L318)
+
+[Microsoft.Owin](https://www.nuget.org/packages/Microsoft.Owin/) uses a nullable SameSite field in most places.
```
Comment (Tratcher) [inline aspnet/SameSite/owin-samesite.md]:
```suggestion
```

**Comment by Rick-Anderson** [comment]
@Tratcher  take a look at `Microsoft.Owin` has its own `SameSite` implementation:`

File: aspnet/SameSite/owin-samesite.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+For the ASP.NET 4.x version of this article, see <xref:samesite/system-web-samesite>.
+
+## API usage with SameSite
+
+`Microsoft.Owin` has its own `SameSite` implementation:
+
+* That is not directly dependent on the one in `System.Web`.
+* It works on all versions targetable by the packages of .NET 4.5 and later.
+* Only the [SystemWebCookieManager](https://github.com/aspnet/AspNetKatana/blob/dev/src/Microsoft.Owin.Host.SystemWeb/SystemWebCookieManager.cs) component directly interacts with the `System.Web` `HttpCookie` class.
+* It depends on the .NET 4.7.2 `System.Web` APIs to enable `SameSite` support, and the patches to change the behavior.
```
[the remark this answers is not in the document]
Comment (Rick-Anderson) [reply aspnet/SameSite/owin-samesite.md]: Changed `It` to

`Microsoft.Owin` depends on the .NET 4.7.2 `System.Web` APIs to enable `SameSite` support, and the patches to change the behavior.

[the remark this answers is not in the document]
Comment (Tratcher) [reply aspnet/SameSite/owin-samesite.md]: No, it's SystemWebCookieManager that depends on .NET 4.7.2 System.Web...

**Pull request merged**